rectools/models/random.py

```diff
@@ -118,3 +118,27 @@ def _recommend_i2i(
         sorted_item_ids_to_recommend: tp.Optional[InternalIdsArray],
     ) -> tp.Tuple[InternalIds, InternalIds, Scores]:
         return self._recommend_u2i(target_ids, dataset, k, False, sorted_item_ids_to_recommend)
+
+    def _get_filtered_item_ids(self, sorted_item_ids_to_recommend: tp.Optional[InternalIdsArray]) -> InternalIdsArray:
+        if sorted_item_ids_to_recommend is not None:
+            item_ids = np.unique(sorted_item_ids_to_recommend)
```

Comment:
`sorted_item_ids_to_recommend` are already after `np.unique`. Why do we do it here?

Reply:
Fixed